provider/pkg/schema/jsschema_util.go

```diff
@@ -34,7 +34,22 @@ func FlattenJSSchema(sch *jsschema.Schema) jsschema.SchemaList {
 		schemas[i] = merged
 	}
 	return schemas
+}
+
+func NormaliseTypes(sch *jsschema.Schema) *jsschema.Schema {
```

Comment:
Seems like a good candidate for unit tests.

Reply:
I've gone with just adding more tests at the overall property type generation level instead.